sys/initrd.c

```diff
@@ -243,16 +239,19 @@ static int initrd_root(mount_t *m, vnode_t **v) {
   return 0;
 }
 
+static vnodeops_t initrd_vops = {.v_lookup = initrd_vnode_lookup,
+                                 .v_read = initrd_vnode_read,
+                                 .v_open = vnode_open_generic,
+                                 .v_close = vnode_close_generic,
+                                 .v_getattr = initrd_vnode_getattr,
+                                 .v_readdir = vnode_op_notsup};
+
 static int initrd_init(vfsconf_t *vfc) {
   unsigned rd_size = ramdisk_get_size();
 
   if (!rd_size)
     return ENXIO;
 
-  initrd_ops.v_lookup = initrd_vnode_lookup;
-  initrd_ops.v_read = initrd_vnode_read;
-  initrd_ops.v_open = vnode_open_generic;
-  initrd_ops.v_getattr = initrd_vnode_getattr;
```

Comment:
If we can't load the ramdisk then `ENXIO` (*no such device or address*) is returned, and mount operation fails.